### DISCUSSION TITLE
Set Static IP - no permission to create directory? [solved]

### DISCUSSION THREAD

**Discussion by rpm8200** [Guilouz/Klipper-Flsun-Speeder-Pad#86] [General]
I do not like DHCP for various reasons, to me it is absolutely not understandable, why there is no simple option in the speederpad/klipper to set an IP. It could be DHCP by default, but why is the only option I have to enter the WLAN Password, but I cant enter the IP, Subnet, gateway, DNS? This really is strange.

Well, I tried to do this like shown at https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad#set-a-static-ip-address
but I get this error right away:
mkdir: cannot create directory ‘/root/speeder_pad’: Permission denied

Can someone tell me what I am doing wrong? Help is very appreciated, because the Adress, which the speederpad is using right now is not even an address of my home network: 192.168.0.10. Also not sure, how this can happen, because my router does not use the 192.168.0 Subnet...? Really strange. My idea was to set it permanently and get it fixed by doing as told in the manual.

**Comment by manwald** [comment]
As you suggest this seems to be permission-related. The root-user should have the permission, though you can try to use 'sudo mkdir...' alternatively.

**Reply by rpm8200** [reply]
ah... sorry!
I missed this one, right? 
https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad#enable-root-access

Or as you said, alternatively sudo...
Sorry and thanks a lot!

**Comment by rpm8200** [comment]
After enabling root access, it was all fine - resolved for me. 
Thanks!
